tests/gracenote_tests.ts

```diff
@@ -36,25 +36,25 @@ function basic(options: TestOptions): void {
     { keys: ['f/4'], duration: '32' },
     { keys: ['g/4'], duration: '32' },
     { keys: ['a/4'], duration: '32' },
-  ].map(f.GraceNote.bind(f));
+  ].map(f.GraceNote.bind(f)) as GraceNote[];
 
-  const gracenotes1 = [{ keys: ['b/4'], duration: '8', slash: false }].map(f.GraceNote.bind(f));
+  const gracenotes1 = [{ keys: ['b/4'], duration: '8', slash: false }].map(f.GraceNote.bind(f)) as GraceNote[];
```

Comment:
Why do we need all these casts? How come these errors don't show up on master or my branch when we are compiling VexFlow?

For example, here we have:
```
[...].map(f.GraceNote.bind(f)) as GraceNote[];
```
The `Factory.GraceNote(noteStruct: GraceNoteStruct): GraceNote` returns a GraceNote, so if we map an array of structs, my TS compiler correctly decides that the return value is `const gracenotes: GraceNote[]`. This can be verified when I mouse over each of those parts of the code (in VS Code, which I assume you are using).

I'm not a fan of all these `as XXX[]` casts if we can avoid them.

Reply:
In case this doesn't work for @rvilarl , maybe it's also because of the older typescript version (that he was at least using some days ago)?